### ISSUE TITLE
Merge the "skip" attribute from included terragrunt files is not supported

### ISSUE THREAD

**Issue by rodrigorfk** [gruntwork-io/terragrunt#3224]
## Describe the bug

The current Terragrunt documentation, specifically the section on the skip attribute [here](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#skip), stipulates that this flag must be explicitly set within Terragrunt modules that are intended to be skipped. This implies that the configuration will not be inherited from include blocks. Despite this being clearly documented, it appears to contradict the merging concept supported by the include block.

From a user's standpoint, it would be beneficial to have the ability to inherit this flag/attribute, similar to how many other attributes are inherited. Is there a specific rationale behind the distinct treatment of this attribute that I may be overlooking?

## Steps To Reproduce

Create a partial terragrunt hcl file

partial.hcl
```hcl
locals {
  module_path_parts = split("/", path_relative_to_include())
  module_base_path  = local.module_path_parts[0]
  module_name       = join("_", slice(local.module_path_parts, 1, length(local.module_path_parts)))
}

skip = !endswith(local.module_name, "_account_layer")
```

Include the partial file in a live module

partial.hcl
```hcl
include {
  path   = find_in_parent_folders("partial.hcl")
  expose = true
}
```

## Expected behavior

As you can see, the `skip` in the partial terragrunt file does not contain a simple `true` or `false`, it contains a more advanced logic that I would like to inherit in included modules, without having to duplicate the logic there, currently that is not possible because the `skip` attribute is not included.

## Nice to haves

- [ ] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version: [v0.59.5](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.59.5)
- OpenTofu/Terraform version: 1.5.5
- Environment details (Ubuntu 20.04, Windows 10, etc.): macOS 14.x

**Comment by yhakbar** [comment]
Hey @rodrigorfk ,

Thanks for opening this issue, and supplying a pull request to address your needs. Your initiative is appreciated!

We will need to take some time to review this logic, and make sure it doesn't impact anything too severely, as it is a breaking change in what's expected from an `include`.

To be transparent, I don't know why `skip`s aren't `include`d right now, but there's likely a good reason, and we want to make sure those are understood before merging anything in.

**Comment by yhakbar** [comment]
The context on this is that `skip` used to be part of a legacy `terragrunt` field that no longer exists. At the time, it didn't make sense to allow the `skip` to be included, but it's totally fine now.

How we handle skips is something that we're working on actively, and is likely to change over time, particularly if the following RFC is accepted:
https://github.com/gruntwork-io/terragrunt/issues/3134

**Comment by yhakbar** [comment]
Fix pre-released in [v0.68.0-beta2024100701](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.68.0-beta2024100701). Will be included in the next minor release.
